lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestFailedEvaluatorEventHandler.cs

```diff
@@ -143,15 +130,13 @@ public void Dispose()
 
             public byte[] Call(byte[] memento)
             {
-                _countdownEvent.Wait();
+                Logger.Log(Level.Info, "Entered FailEvaluatorTask");
+
+                // need to sleep to allow Java code to get heartbeat with information about task
+                Thread.Sleep(100);
```

Comment:
Yes, the time for task information to propagate to Java code is more than time for .NET code to fail.

I've just tried old version of the test, with `CountdownEvent` and `RunningTask`, and it passes both locally and [on AppVeyor](https://ci.appveyor.com/project/tcNickolas/reef/build/136-REEF-1388-t/tests). Most likely I've done this change when trying to fix tests before deadlock in reporting was resolved. 
I'll revert this test (which also addresses the other comments on it)